models/ios/evpn_global/ios_evpn_global.yaml

```diff
@@ -0,0 +1,89 @@
+module: ios_evpn_global
+short_description: Resource module to configure L2VPN EVPN.
+description: This module provides declarative management of L2VPN EVPN on Cisco IOS network
+  devices.
+version_added: 1.0.0
+author: Padmini Priyadarshini Sivaraj (@PadminiSivaraj)
+notes:
+  - Tested against Cisco IOSl2 device with Version 15.2 on VIRL.
+  - Starting from v2.5.0, this module will fail when run against Cisco IOS devices that do
+    not support L2VPN EVPN. The offline states (C(rendered) and C(parsed)) will work as expected.
+  - This module works with connection C(network_cli).
+    See U(https://docs.ansible.com/ansible/latest/network/user_guide/platform_ios.html)
+options:
+  config:
+    description: A dictionary of L2VPN Ethernet Virtual Private Network (EVPN) configuration
+    type: dict
+    suboptions:
+      default_gateway:
+        description: Default gateway parameters
+        type: dict
+        suboptions:
+          advertise:
+            description: Advertise Default Gateway MAC/IP routes
+            type: bool
+      flooding_suppression:
+        description: Suppress flooding of broadcast, multicast, and/or unknown unicast packets
+        type: dict
+        suboptions:
+          address_resolution:
+            description: Suppress flooding of Address Resolution and Neighbor Discovery Protocol packets
+            type: dict
+            suboptions:
+              disable:
+                description: Disable flooding suppression
+                type: bool
+      ip:
+        description: IP parameters
+        type: dict
+        suboptions:
+          local_learning:
```

Comment:
if there is no enabled option we can simplify the model to just the top level key

Reply:
Since evpn_global and evpn_evi plugin share the same parameter, and evpn_evi has both enable and disable option (this plugin doesn't), I didn't simplify this key here just to be consistent.